website/source/api/secret/ad/index.html.md

```diff
@@ -191,6 +191,222 @@ $ curl \
   "username": "my-application"
 }
 ```
+
+## Library management
+
+The `library` endpoint configures the sets of service accounts that Vault will offer for check-out.
+
+### Parameters
+
+* `name` (string: "", <required>): The name of the set of service accounts.
+* `service_account_names` (string: "", or list: [] <required>): The names of all the service accounts that can be 
+checked out from this set. These service accounts must only be used by Vault, and may only be in one set. These 
+service accounts must already exist in Active Directory.
+* `ttl` (duration: "24h", <optional>): The maximum amount of time a single check-out lasts before Vault 
+automatically checks it back in. Defaults to 24 hours. Setting it to zero reflects an unlimited lending period.
+* `max_ttl` (duration: "24h", <optional>): The maximum amount of time a check-out last with renewal before Vault 
+automatically checks it back in. Defaults to 24 hours. Setting it to zero reflects an unlimited lending period.
+* `disable_check_in_enforcement` (bool: false, <optional>): Disable enforcing that service accounts must be 
+checked in by the entity or client token that checked them out. Defaults to false.
+
+When adding a service account to the library, Vault verifies it already exists in Active Directory.
+
+| Method   | Path                    |
+| :------- | :---------------------- |
+| `LIST`   | `/ad/library`           |
+| `POST`   | `/ad/library/:set_name` |
+| `GET`    | `/ad/library/:set_name` |
+| `DELETE` | `/ad/library/:set_name` |
+
+### Sample Post Request
+
+```
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request POST \
+    --data @payload.json \
+    http://127.0.0.1:8200/v1/ad/library/accounting-team
+```
+
+### Sample Post Payload
+
+```json
+{
+  "service_account_names": ["fizz@example.com", "buzz@example.com"],
+  "ttl": "10h",
+  "max_ttl": "20h",
+  "disable_check_in_enforcement": false
+}
+```
+
+### Sample Get Response
+
+```json
+{
+  "service_account_names": ["fizz@example.com", "buzz@example.com"],
+  "ttl": "10h",
+  "max_ttl": "20h",
+  "disable_check_in_enforcement": false
+}
+```
+
+### Sample List Response
+
+Performing a `LIST` on the `/ad/library` endpoint will list the names of all the sets of service accounts Vault contains.
+
+```json
+[
+  "accounting-team"
+]
+```
+
+## Check-out management
+
+These endpoints help manage check-outs.
+
+### Check a credential out
+
+Returns a `200` if a credential is available, and a `400` if no credential is available.
+
+* `name` (string: "", <required>): The name of the set of service accounts.
+* `ttl` (duration: "", <optional>): The maximum amount of time a check-out lasts before Vault 
+automatically checks it back in. Defaults to 24 hours. Setting it to zero reflects an unlimited lending period.
```

Comment:
I'm a little confused by this.  It says _Defaults to 24 hours_ then turns around and says _Defaults to the set's `ttl`_.  Wondering if there's a way to make this more clear?

Reply:
Whoops, LOL, yes, it only defaults to one of those things. Yep, I'll fix that.